### DISCUSSION TITLE
Run setup script in init hook?

### DISCUSSION THREAD

**Discussion by alexispurslane** [89luca89/distrobox#1250] [Q&A]
I have a setup script that compiles a couple packages I need from source and installs them, clones my dotfiles and unstow's them, builds doom emacs, etc, and I'd like to make that script an init hook in my `distrobox.ini` so that when I run `distrobox assemble create` it runs those steps as part of the process of creating my development environment container, after all the packages have installed.

I've tried putting each of the commands in the set up script individually in init_hooks, but for some reason it kept failing (something to do with the way init_hooks are parsed and combined), so I'd really like to have them bundled up in a discrete script instead, so that they're executed correctly. Is there any way to make that script accessible to a distrobox's init hooks?

**Comment by alexispurslane** [comment]
![image](https://github.com/89luca89/distrobox/assets/1920151/9ad42b47-398c-4170-b7be-f3ce0a736bec)
Currently it just gives this.

**Comment by MarkusOstermayer** [comment]
You don't need to recreate your script line-by-line using `init_hooks` but can provide `init_hooks` a path to the script. Just make sure that everything the script needs in order to run is already installed in the distrobox.


According to [https://distrobox.it/#configure-distrobox](https://distrobox.it/#configure-distrobox) `container_init_hook` is capable of working with shell-scripts and internally distrobox-create's `--init-hooks`-flag uses `container_init_hook`:
https://github.com/89luca89/distrobox/blob/4c38d6b3f9bc833e58a435471158935824727805/distrobox-create#L415-L420

**Comment by happymab** [comment]
You can use --volume option to make the script available inside distrobox. 
--volume /path/to/your/script:/path/inside/container:ro
